cub/benchmarks/bench/reduce/sum.cu

```diff
@@ -25,11 +25,17 @@
  *
  ******************************************************************************/
 
+// This benchmark is intended to cover redux instructions on Ampere+ architectures. It specifically uses
+// cuda::std::plus<> instead of a user-defined operator, which CUB recognizes to select an optimized code path.
+
+// Tuning parameters found for signed integer types apply equally for unsigned integer types
+
 #include <nvbench_helper.cuh>
 
 // %RANGE% TUNE_ITEMS_PER_THREAD ipt 7:24:1
 // %RANGE% TUNE_THREADS_PER_BLOCK tpb 128:1024:32
 // %RANGE% TUNE_ITEMS_PER_VEC_LOAD_POW2 ipv 1:2:1
 
-using op_t = ::cuda::std::plus<>;
+using value_types = push_back_t<all_types /*, __half, __nv_bfloat16*/>;
```

Comment:
important: this will lead to tuning issues going forward. Currently, we have `all_types` that has many types. During tuning, it'll be changed to contain only one type. If we append this list in the benchmark itself, we'll end up compiling thousands of `.variant` executables for a given `T` *and* half / bfloat. Instead, I'd suggest to have a named list in nvbech so that when tuning is compiled, type list contains exactly one type and save us build time.

Reply:
Right, because `all_types` is redefined to a single type during tuning. Good catch! I added TODO comments to add the half and bfloat16 types at some point.